### DISCUSSION TITLE
Hide the widget from the dashboard

### DISCUSSION THREAD

**Discussion by romaldyminaya** [filamentphp/filament#1683] [Help] [answered]
Hi,

When I create a widget in the application, it gets display in the main dashboard by default. But I just want the widget to be displayed in certain pages.

How do I hide it form the main dashboard ?

Thanks in advanced :)

**Comment by ehs4nm** [comment]
I have found a workaround about this problem. maybe there are better ways but here is my solution:
Add this in your widget class and control which roles can view the widget, off course you can customize it in a way to control more detailed way.
```
    public static function canView(): bool
    {
        return auth()->user()->hasRole('super_admin');
    }
```

**Comment by danharrin** [comment, marked as answer]
If you don't want the widget on the dashboard page, don't put it in the `/Widgets` directory. Create a new `/Pages/PageName/Widgets` directory & namespace instead maybe.

@ehs4nm is correct though, if you want to control who can view the widget then you can use `canView()`, but that doesn't control registration.

**Reply by romaldyminaya** [reply]
Good, I'll create different folders and namespaces and then use the canView as @ehs4nm mentioned to control other kngotmstions. Thanks 😊
